iOS 11 and iPhone X visual fixes

nicorsm opened the pull request:
In this pull request:
- fixed status bar position in country list view controller (iPhone X)
- adjusted the position of the closing button (X) in the signup view controller (iOS 11, iPhone X)
- fixed alignment of user/group name with avatar in video calls (iOS 11, iPhone X)
- fixed a visual issue with the navigation bar while accessing to conversation pictures from the search button of conversation view (iOS 11, iPhone X)
- fixed bottom alignment of emoji keyboard (iPhone X)
- fixed an issue where the background color was not updated in conversations after changing theme (iPhone X)
- changed style of the "create group"/"add participants" button on the Add Participants view controller, as suggested by @eike96. Now the button has rounded corners and margin. (iOS 11, iPhone X)
- fixed bottom alignment of record button in record keyboard (iPhone X)
- fixed missing background color in Canvas View Controller (iPhone X)
- fixed top and bottom alignments for alert views (iPhone X)
- removed default grey selection on devices list tab of user profile (iOS 11, iPhone X)
- fixed top and bottom margins on user client view controller (iPhone X)
- changed UI Tests files
- minor alignment fix to Chat Heads (iPhone X)
- adjusted bottom margin while composing drafts (iPhone X)
- handled keyboard appearance on new draft textviews - title and content (iOS 11, iPhone X)

nicorsm commented:
![wire 2017-10-13 at 17 40 31](https://user-images.githubusercontent.com/2906234/31613177-959e3258-b282-11e7-9920-1e90afa94532.png)
Bottom button prototype by @eike96

mikeger commented on WireExtensionComponents/Views/Buttons/IconButton.m:
```diff
@@ -340,13 +346,20 @@ - (void)updateTintColor
 
+- (void)updateCustomCornerRadius
+{
+    if(self.roundCorners) {
```
There should be an `else` case to remove the rounded corners.

mikeger commented:
@nicorsm this PR has 2 parts: 

1. Some general UI fixes.
2. iPhone X improvements.

For 1: I would vote to separate the fixes in 2 PRs, since it's hard sometimes to understand from the code what the fix is.
For 2: I would propose to sit and discuss afterwards about the iPhone X compatibility, may be we can think how to improve it and may be make more general.

mikeger commented:
@nicorsm I think the tests are failed.

nicorsm commented:
We could split the PR in this way:

**iPhone X**
- https://github.com/wireapp/wire-ios/pull/1359/commits/1a994700228a09f5d09f83dcfb1a2b9e01262a2c fixed audio record button, canvas, actionsheet and archived list margins for iPhone X
- https://github.com/wireapp/wire-ios/pull/1359/commits/a6c499112581cf398eaea3b1b99775439d076dd9 fixed canvas background color and emoji keyboard insets
fixed canvas background color and emoji keyboard insets
- https://github.com/wireapp/wire-ios/pull/1359/commits/7805301652a1a6aa14a4db2615e29ce7fb12feef adjusted insets for Confirm Asset, Profile and Profile Client view controllers. deleted default cell behavior on selection for Participant Device cell
- https://github.com/wireapp/wire-ios/pull/1359/commits/7f6aae417d9b0e722b3c612973bc1aed12c96f4c fixed bottom margin on self profile picture, top margin on country code table view controller
- https://github.com/wireapp/wire-ios/pull/1359/commits/d14104111372f8ae64366e67635a27f76bc6575f fixed keyboard bottom margin on iPhone X while creating a new draft

**iOS 11**
- https://github.com/wireapp/wire-ios/pull/1359/commits/9f93c100b91b639ab7e18cde4297a7cb929f3db4 Modified UI tests reference images in order to be compliant with new sizes
- https://github.com/wireapp/wire-ios/pull/1359/commits/0c4bbcddc12576c3f212a376a4e26e639fb3466b fixed close button on registration, text alignment on voice overlay
- https://github.com/wireapp/wire-ios/pull/1359/commits/30d6c0df30479a4b01a08453ac5debb79283590d fixed navigation bar visual issues on Conversation Images view controller
- https://github.com/wireapp/wire-ios/pull/1359/commits/9c564b6f3523227f118ce7f77e0886989ef29bec new Add People button on Add Participants view controller
- https://github.com/wireapp/wire-ios/pull/1359/commits/f353a8b0f18556b1cc187f93917c9370973d94aa fixed keyboard appearance on Message Compose + insets for Chat Heads

mikeger commented:
@nicorsm thanks, I guess it's not necessary to actually split the PR now, please check the tests and comments and let's merge it 🎉

nicorsm commented:
✅ Green light from Design 🎉

typfel commented:
The tests will fail on CI because of the dot pattern in calling UI. So I guess now is good time to figure out why the appear on DEV but not on CI.

nicorsm commented:
@typfel with the tests I made on the last commit, I don't see the dots while testing locally but, actually, tests are failing on CI. 🤷‍♂️

typfel commented:
@nicorsm I see in the diff that the new screenshot reference has the dots. I'll debug why they appear on dev builds.

nicorsm commented:
@typfel In the meanwhile, I could restore the wrong snapshots to the previous version, so CircleCI tests can succeed.

nicorsm commented:
...It doesn't solve the issue because there are some little misalignments with the title label.
